be/src/tools/builder_scanner_memtable.cpp

```diff
@@ -0,0 +1,346 @@
+
+        _params.expr_of_dest_slot.emplace(i, expr);
+        _params.src_slot_ids.push_back(_tablet->num_columns() + i);
+    }
+
+    // _params.__isset.expr_of_dest_slot = true;
+    _params.__set_dest_tuple_id(TUPLE_ID_DST);
+    _params.__set_src_tuple_id(TUPLE_ID_SRC);
+}
+
+void BuilderScannerMemtable::build_scan_ranges(std::vector<TBrokerRangeDesc>& ranges,
+                                               const std::vector<std::string>& files) {
+    LOG(INFO) << "build scan ranges for files size:" << files.size() << " file_type:" << _file_type;
+    for (const auto& file : files) {
+        TBrokerRangeDesc range;
+        range.start_offset = 0;
+        range.size = -1;
+        range.format_type = TFileFormatType::FORMAT_PARQUET;
+        range.splittable = true;
+
+        range.path = file;
+        range.file_type = _isHDFS ? TFileType::FILE_HDFS : TFileType::FILE_LOCAL;
+        ranges.push_back(range);
+    }
+
+    if (!ranges.size()) LOG(FATAL) << "cannot get valid scan file!";
+}
+
+void BuilderScannerMemtable::doSegmentBuild(const std::vector<std::string>& files) {
+    vectorized::VBrokerScanNode scan_node(&_obj_pool, _tnode, *_desc_tbl);
+    scan_node.init(_tnode);
+    auto status = scan_node.prepare(&_runtime_state);
+    if (!status.ok()) LOG(FATAL) << "prepare scan node fail:" << status.to_string();
+
+    // set scan range
+    std::vector<TScanRangeParams> scan_ranges;
+    {
+        TScanRangeParams scan_range_params;
+
+        TBrokerScanRange broker_scan_range;
+        broker_scan_range.params = _params;
+        build_scan_ranges(broker_scan_range.ranges, files);
+        scan_range_params.scan_range.__set_broker_scan_range(broker_scan_range);
+        scan_ranges.push_back(scan_range_params);
+    }
+
+    scan_node.set_scan_ranges(scan_ranges);
+    status = scan_node.open(&_runtime_state);
+    if (!status.ok()) LOG(FATAL) << "open scan node fail:" << status.to_string();
+
+    // std::unique_ptr<RowsetWriter> rowset_writer;
+    PUniqueId load_id;
+    load_id.set_hi(1);
+    load_id.set_lo(1);
+    int64_t transaction_id = 1;
+
+    // delta writer
+    TupleDescriptor* tuple_desc = _desc_tbl->get_tuple_descriptor(TUPLE_ID_DST);
+    WriteRequest write_req = {_tablet->tablet_meta()->tablet_id(),
+                              _tablet->schema_hash(),
+                              WriteType::LOAD,
+                              transaction_id,
+                              _tablet->partition_id(),
+                              load_id,
+                              tuple_desc,
+                              &(tuple_desc->slots())};
+
+    DeltaWriter* delta_writer = nullptr;
+    DeltaWriter::open(&write_req, &delta_writer, load_id);
+    status = delta_writer->init();
+    if (!status.ok()) LOG(FATAL) << "delta_writer init fail:" << status.to_string();
+
+    std::filesystem::path segment_path(std::filesystem::path(_build_dir + "/segment"));
+    std::filesystem::remove_all(segment_path);
```

Comment:
Nit::
Can we guarantee that the `segment_path` is always a new empty directory ? 
If so it had better to failed rather than `remove all` when there are files under this path.